fs/inode/fs_files.c

```diff
@@ -525,6 +525,11 @@ int nx_dup2_from_tcb(FAR struct tcb_s *tcb, int fd1, int fd2)
   ret = file_dup2(&list->fl_files[fd1 / CONFIG_NFILE_DESCRIPTORS_PER_BLOCK]
                                  [fd1 % CONFIG_NFILE_DESCRIPTORS_PER_BLOCK],
                   filep);
+
+#ifdef CONFIG_FDSAN
+  filep->f_tag = file.f_tag;
```

Comment:
should we fix freopen instead changing dup?

Reply:
If you make any changes, you may ping me to re-test.

Reply:
Sure.

Reply:
@xiaoxiang781216 
Freopen will call dup2 and dup2 is also an API for the user layer. 
If we only modify freopen's implementation, users will still encounter errors when using dup2 directly.

Fdsan can be actively or passively used by users.
If the user passively uses fdsan. we do not expect that dup2 will change f_tag.
If the user actively uses fdsan, they will actively call the fdsan API to change the f_tag.

Reply:
Ok, I misunderstand the code, the change looks good.